test/test_all.py

```diff
@@ -0,0 +1,227 @@
+# This file is part of libraries-repository-engine.
+#
+# Copyright 2021 ARDUINO SA (http://www.arduino.cc/)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# You can be released from the requirements of the above licenses by purchasing
+# a commercial license. Buying such a license is mandatory if you want to
+# modify or otherwise use the software for commercial activities involving the
+# Arduino software without disclosing the source code of your own applications.
+# To purchase a commercial license, send an email to license@arduino.cc.
+#
+
+import string
+import re
+import hashlib
+import json
+import pathlib
+import platform
+import typing
+
+import invoke.context
+import pytest
+
+test_data_path = pathlib.Path(__file__).resolve().parent.joinpath("testdata")
+
+
+def test_all(run_command, working_dir):
+    working_dir_path = pathlib.Path(working_dir)
+    configuration = {
+        "BaseDownloadUrl": "http://www.example.com/libraries/",
+        "LibrariesFolder": working_dir_path.joinpath("libraries").as_posix(),
+        "LogsFolder": working_dir_path.joinpath("logs").as_posix(),
+        "LibrariesDB": working_dir_path.joinpath("libraries_db.json").as_posix(),
+        "LibrariesIndex": working_dir_path.joinpath("libraries", "library_index.json").as_posix(),
+        "GitClonesFolder": working_dir_path.joinpath("gitclones").as_posix(),
+        # I was unable to get clamdscan working in the GitHub Actions runner, but the tests should pass with this set to
+        # False when run on a machine with ClamAV installed.
```

Comment:
As we were discussing in our chats, being this application something that has to run on a well defined environment, could make sense to include in this repo a `Dockerfile` with a replica of the destination linux environment, included all the tools that this engine uses (like ClamAV)